amethyst_assets/src/prefab/mod.rs

```diff
@@ -352,15 +352,29 @@ where
     ) -> Result<Handle<A>, PrefabError> {
         let handle = match *self {
             AssetPrefab::Handle(ref handle) => handle.clone(),
-            AssetPrefab::File(ref name, ref format, ref options) => system_data.0.load(
+            AssetPrefab::File(..) => unreachable!(),
+        };
+        system_data.1.insert(entity, handle.clone()).map(|_| handle)
+    }
+
+    fn load_sub_assets(&mut self, progress: &mut ProgressCounter, system_data: &mut Self::SystemData) -> Result<bool, PrefabError> {
+        let handle = if let AssetPrefab::File(ref name, ref format, ref options) = *self {
+            Some(system_data.0.load(
                 name.as_ref(),
                 format.clone(),
                 options.clone(),
-                (),
+                progress,
```

Comment:
That's probably what caused my gltf to be considered done loading when it wasn't
(which was still happening yesterday)

Reply:
Yep. I went through all the prefabs and checked, and this one was "wtf, this is bad"